exe-nix-buildkite/Main.hs

```diff
@@ -62,6 +63,40 @@ withTime label k = do
   where
     click = getTime Monotonic
 
+nixInstantiate :: String -> IO [String]
+nixInstantiate jobsExpr = withTime "nix-instantiate" (Prelude.lines <$> readProcess "nix-instantiate" [ jobsExpr ] "")
+
+nixBuildDryRun :: String -> IO [String]
+nixBuildDryRun jobsExpr = withTime "nix-build --dry-run" do
+  (_stdin, _stdout, stderrHndl, _prchndl) <- createProcess $ (proc "nix-build" ["--dry-run", jobsExpr]) { std_err = CreatePipe }
+
+  inputLines <- Prelude.lines <$> case stderrHndl of
+    Just hndl -> hGetContents hndl
+    Nothing -> pure []
+  -- See Note: [nix-build --dry-run output]
+  let stripLeadingWhitespace = dropWhile (==' ')
+  let theseLine = List.isPrefixOf "these"
+
+  -- dump the output to stderr
+  mapM_ (hPutStrLn stderr) inputLines
+
+  pure $ map stripLeadingWhitespace . takeWhile (not . theseLine) . drop 1 $ dropWhile (not . theseLine) inputLines
+
+-- Note: [nix-build --dry-run output]
+-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+-- The output of `nix-build --dry-run` looks like this (on stderr! not stdout):
```

Comment:
If we were using the flake for this we could use `nix build --json`, but :shrug:

Reply:
Well that's orthogonal to flakes we can also do `nix build --dry-run --json -f ...`. I thought I would use this one just so we don't break if a user has the experimental cli disabled